cmd/istio_ca/main.go

```diff
@@ -85,6 +89,8 @@ func init() {
 	flags.StringVar(&opts.namespace, "namespace", "",
 		"Select a namespace for the CA to listen to. If unspecified, Istio CA tries to use the ${"+namespaceKey+"} "+
 			"environment variable. If neither is set, Istio CA listens to all namespaces.")
+	flags.StringVar(&opts.caNamespace, "ca-namespace", "istio-system", "Specify whether the current CA is running at.")
```

Comment:
We fix one bug and we introduce UX limitation. With the current code and yaml templates, users can do a sed '/istio-system/kube-system' and still have a cluster wide installation. With this change, they will have to also start the CA with the flag --ca-namespace "kube-system". It is an acceptable trade-off as long as this gets documented. Even better would be to use POD_NAMESPACE for this.

Please replace "Specify whether the current CA is running at." with "Namespace where the Istio CA pods is running. If not specified, it defaults to POD_NAMESPACE".

In 0.3, we will need to swap the flags, since all controller use "namespace" mapped to POD_NAMESPACE, to indicate where the controller is running, and other flag names to indicate where they listen to. This will be a backward incompatible change, thus my recommendation was to do it properly in 0.2 , if we must fix the critical bug.

Reply:
Thanks Andra for your input.

Comment of ca-namespace flag is updated.

Currently it's NAMESPACE, it's just one line change for me to use POD_NAMESPACE, do you think it is safe to change to POD_NAMESPACE for now?

Agreed on 0.3 vision.

Reply:
How about "Specify the namespace where Istio CA root CA is stored. Will not be used if explicit file or other storage mechanism is specified"

And rename it to 'istio-ca-storage-namespace' - to make it very clear it is specific to the CA storage (and avoid mixing it up with what Andra is concerned about )

Reply:
Comments and flag name are both updated